core/src/actors/session/handlers.rs

```diff
@@ -77,6 +81,22 @@ impl StreamHandler<BytesMut, Error> for Session {
                     ) => {
                         peer_discovery_peers(&peers);
                     }
+                    //////////////
+                    // GET DATA //
+                    //////////////
+                    (_, SessionStatus::Consolidated, Command::GetData(GetData { inventory })) => {
+                        for elem in inventory {
+                            match elem {
+                                InvElem::Block(hash)
+                                | InvElem::Tx(hash)
+                                | InvElem::DataRequest(hash)
+                                | InvElem::DataResult(hash) => {
+                                    self.send_message(create_block_msg(hash));
```

Comment:
:heart: yesterday I tried to do this exactly and found no way of doing it, I'm going to copy this from you :smile:

Reply:
Renamed InvElem to InvVector :)